File: Java/rabbit-mq/src/main/java/core/rabbitmq/HelloController.java
package core.rabbitmq;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;

@RestController
public class HelloController {
    @GetMapping("/hello")
    public HashMap<String, String> hello(@RequestParam String name) {
        HashMap<String, String> hashMap = new HashMap<>();
        hashMap.put("key", "Hello World: " + name);
        return hashMap;
    }
}

File: Project/kotlin-jpa/src/main/kotlin/com/jpa/kotlinjpa/controller/HelloController.kt
package com.jpa.kotlinjpa.controller

import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@Validated
@RestController
class HelloController {

    data class HelloDto(
            var name: String?
    ) {
        init {
            if (name.isNullOrBlank()) {
                this.name = null
            }
        }
    }

    @GetMapping("/hello")
    fun helloRequestParam(
            @RequestParam name: String?
    ): Map<String, String> {
        return mapOf("message" to "hello $name")
    }

    @GetMapping("/hello/{name}")
    fun helloPathVariable(
            @PathVariable name: String?
    ): Map<String, String> {
        return mapOf("message" to "hello $name")
    }

    @PostMapping("/hello")
    fun helloRequestBody(
            @RequestBody dto: HelloDto
    ): Map<String, String> {
        return mapOf("message" to "hello ${dto.name}")
    }
}
